ex3/ex3.c

```diff
@@ -9,6 +9,16 @@
 int main(int argc, char* argv[])
 {
     // Your code here
-
+int rc = fork();
+    if (rc < 0) {
+        fprintf(stderr, "fork failed\n");
+        exit(1);
+        } 
+    else if (rc == 0) {
+        printf("hello, I'm a child\n");
+    } else {
+        int wc = waitpid(rc, NULL, 0);
```

Comment:
When compiling this code, this line throws the warning `implicit declaration of function ‘waitpid’ [-Wimplicit-function-declaration]` - this won't prevent the code from compiling, however implicit declaration warnings occur when you haven't included the header which defines how that function should behave. What this means is that the compiler will have no way of verifying whether you're using the function correctly at compile time, which can lead to unexpected behavior when the program is actually run. Just something worth keeping in mind with C.